spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -510,6 +508,69 @@ public void init(@Metric(name="instances") Counter instances) {
 }
 ----
 
+=== Registering metrics dynamically
+In addition to declaring metrics via annotations, it is possible to dynamically (un)register metrics by 
+calling methods of a `MetricRegistry` object. While using annotations is generally the preferred approach,
+registering metrics dynamically can be useful in cases when the final list of metrics is not known in 
+advance (when the application is being coded) or when there are too many metrics similar in a way
+that it will be more practical to register them in a `for` loop, for example, than introducing 
+lots of annotations in the code. The two approaches can also be combined if necessary.
```

Comment:
Should we also comment that it is often not a good practice to create millions of metrics as the downstream time series databases that need to store the metrics may not deal with that well?

Reply:
Yes. I added that to the intro section to the programming model, it feels more appropriate there